File: common/src/main/java/net/superricky/tpaplusplus/commands/accept/TPAAcceptCommand.java
package net.superricky.tpaplusplus.commands.accept;

import com.mojang.brigadier.CommandDispatcher;
import com.mojang.brigadier.exceptions.CommandSyntaxException;
import net.minecraft.commands.CommandSourceStack;
import net.minecraft.commands.arguments.EntityArgument;
import net.minecraft.server.level.ServerPlayer;
import net.superricky.tpaplusplus.config.Config;
import org.apache.commons.lang3.NotImplementedException;

import static net.minecraft.commands.Commands.argument;
import static net.minecraft.commands.Commands.literal;

public class TPAAcceptCommand {

    public static void onRegisterCommandEvent(CommandDispatcher<CommandSourceStack> dispatcher) {
        dispatcher.register(literal(Config.TPAACCEPT_COMMAND_NAME.get())
                .executes(context -> acceptMostRecentTPA(context.getSource()))
                .then(argument("player", EntityArgument.player())
                        .executes(context -> acceptTPASpecified(context.getSource(), EntityArgument.getPlayer(context, "player")))));
    }

    private static int acceptMostRecentTPA(CommandSourceStack source) throws CommandSyntaxException, NotImplementedException {
        AcceptTPA.acceptTeleportRequest(source.getPlayerOrException());
        return 1;
    }

    private static int acceptTPASpecified(CommandSourceStack source, ServerPlayer sender) throws CommandSyntaxException, NotImplementedException {
        AcceptTPA.acceptTeleportRequest(source.getPlayerOrException(), sender);
        return 1;
    }

    private TPAAcceptCommand() {
    }
}

File: common/src/main/java/net/superricky/tpaplusplus/commands/accept/TPAAcceptCommand.kt
package net.superricky.tpaplusplus.commands.accept

import com.mojang.brigadier.CommandDispatcher
import com.mojang.brigadier.context.CommandContext
import com.mojang.brigadier.exceptions.CommandSyntaxException
import net.minecraft.commands.CommandSourceStack
import net.minecraft.commands.Commands
import net.minecraft.commands.arguments.EntityArgument
import net.minecraft.server.level.ServerPlayer
import net.superricky.tpaplusplus.commands.accept.AcceptTPA.acceptTeleportRequest
import net.superricky.tpaplusplus.config.Config
import org.apache.commons.lang3.NotImplementedException

object TPAAcceptCommand {
    fun onRegisterCommandEvent(dispatcher: CommandDispatcher<CommandSourceStack>) {
        dispatcher.register(
            Commands.literal(Config.TPAACCEPT_COMMAND_NAME.get())
                .executes { context: CommandContext<CommandSourceStack> ->
                    acceptMostRecentTPA(
                        context.getSource()
                    )
                }
                .then(
                    Commands.argument("player", EntityArgument.player())
                        .executes { context: CommandContext<CommandSourceStack> ->
                            acceptTPASpecified(
                                context.getSource(),
                                EntityArgument.getPlayer(context, "player")
                            )
                        }
                )
        )
    }

    @Throws(CommandSyntaxException::class, NotImplementedException::class)
    private fun acceptMostRecentTPA(source: CommandSourceStack): Int {
        acceptTeleportRequest(source.playerOrException)
        return 1
    }

    @Throws(CommandSyntaxException::class, NotImplementedException::class)
    private fun acceptTPASpecified(source: CommandSourceStack, sender: ServerPlayer): Int {
        acceptTeleportRequest(source.playerOrException, sender)
        return 1
    }
}
